openverse_catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -8,266 +8,172 @@
 
 Notes:                  https://www.europeana.eu/api/v2/search.json
 """
-
 import argparse
 import logging
 from datetime import datetime, timedelta, timezone
 
+import common
 from airflow.models import Variable
 from common.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.image import ImageStore
-from requests.exceptions import JSONDecodeError
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
-logging.basicConfig(
-    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
-)
 logger = logging.getLogger(__name__)
+logging.getLogger(common.urls.__name__).setLevel(logging.WARNING)
+
+
+class EuropeanaRecordBuilder:
+    """
+    A small class to contain the record building functionality
+    and simplify testing a bit.
+    """
+
+    def get_record_data(self, data: dict) -> dict:
+        record = {
+            "foreign_landing_url": self.get_foreign_landing_url(data),
+            "image_url": data.get("edmIsShownBy")[0],
+            "foreign_identifier": data.get("id"),
+            "meta_data": self.get_meta_data_dict(data),
+            "title": data.get("title")[0],
```

Comment:
For both of these `get(<x>)[0]` calls, it's possible `get` could return `None`, which would cause the indexing into `None` to raise an error. Could we add defaults or extra checking steps for those two fields here?

Reply:
I considered this when writing it, but if either `title` or `image_url` are missing for something, wouldn't we want it to fail out? Is there a different way to fail it out that works better with Airflow's expectations?

Reply:
I think we'd want to skip this record in that case, but not necessarily fail the entire ingestion run. A lot of our providers operate under the assumption that attributes missing from a particular record, even if necessary, would just be ignored and processing would continue. I feel like that might be the best approach here: log that something which was expected to be present was missing, then return `None` from the `get_record_data` function.

Reply:
Right on, makes sense! I wonder if we'd want some kind of "incomplete record store" or something, but a discussion for the a different day :slightly_smiling_face: 

Do you happen to know the complete list of fields that are required for a particular record to be valid? Maybe this is something the base class could handle instead by verifying that all required fields are truthy (`not None` and `!= ""`, I suppose)?

Reply:
@obulat responded well to this in [another comment on this PR](https://github.com/WordPress/openverse-catalog/pull/821#discussion_r1004463445), and I added a note there too about recently added documentation 😄